graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/federation/FederatedSchemaGeneratorHooks.kt

```diff
@@ -44,12 +51,13 @@ import kotlin.reflect.full.findAnnotation
  */
 open class FederatedSchemaGeneratorHooks(private val federatedTypeRegistry: FederatedTypeRegistry) : SchemaGeneratorHooks {
     private val directiveDefinitionRegex = "(^#.+$[\\r\\n])?^directive @\\w+.+$[\\r\\n]*".toRegex(setOf(RegexOption.MULTILINE, RegexOption.IGNORE_CASE))
-    private val customDirectivesRegex = "\\s*@(?!extends)(?!external)(?!key)(?!provides)(?!requires)(?!deprecated)\\w+(?:\\(.*(?=\\))\\))?".toRegex(
-        setOf(RegexOption.MULTILINE, RegexOption.IGNORE_CASE))
     private val scalarDefinitionRegex = "(^#.+$[\\r\\n])?^scalar (_FieldSet|_Any)$[\\r\\n]*".toRegex(setOf(RegexOption.MULTILINE, RegexOption.IGNORE_CASE))
     private val emptyQueryRegex = "^type Query \\{$\\s+^\\}$\\s+".toRegex(setOf(RegexOption.MULTILINE, RegexOption.IGNORE_CASE))
```

Comment:
We just need to update the `directiveDefinitionRegex` and `scalarDefinitionRegex` regex to get a passing build so that they work with the new comment format to use `"Comment"` instead of `#Comment`

This PR would allow us to remove `directiveDefinitionRegex` all together https://github.com/graphql-java/graphql-java/pull/1756

Reply:
since you are using directive filter do we still need the `directiveDefinitionRegex`? I think this would have to handle both single and multiline comments

Reply:
Correct, there is no need for the directive regex now

Reply:
If it is no longer needed please remove it (declaration at line 54 and referenced and line 94).

Reply:
Actually, I misspoke. We can not remove this regex unless this PR is merged and released. https://github.com/graphql-java/graphql-java/pull/1756